x/merkledb/db.go

```diff
@@ -497,6 +497,12 @@ func (db *merkleDB) GetMerkleRoot(ctx context.Context) (ids.ID, error) {
 
 // Assumes [db.lock] is read locked.
 func (db *merkleDB) getMerkleRoot() ids.ID {
+	// if the nil key root is not required, pretend the trie's root is the nil key node's child
+	if db.root.valueDigest.IsNothing() && len(db.root.children) == 1 {
```

Comment:
using valueDigest here, because that is the only one set when doing proof validation